### PR TITLE
Event image gallery ui

### PR DESCRIPTION
**Pull request by mehdirajani5**
Event images work is completed.

1)which event have their own image gallery, they can add new images related to their event.
Backend and frontend (Both are completed sucessfully).

### CODE REVIEW

File: api/event/event.model.js [hunk 1]
```diff
@@ -38,6 +38,11 @@ const eventSchema = mongoose.Schema(
             type: Date,
             required: true,
             default: Date.now
+        },
+        eventImages:
```
Comment (AhsanAyaz) [inline api/event/event.model.js]: Remove `eventImages`

File: api/images/images.router.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+request object: expects a json object of type { object }
+response type: sends a json object of type { "image": object }. 
+*/
+addImage = function(expressInstance)
+{
+    expressInstance.post('/image/new', (req, res) => {
+        let image =  req.body;
+        ImageModel.create(image, function(err , image){
+            if(err)
+            {
```
Comment (AhsanAyaz) [inline api/images/images.router.js]: Spacing issues

File: api/images/images.router.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                res.status(401).send("Unauthorized");
+            }
+            else
+            {
+                
+                res.json({ "image": image });
```
Comment (AhsanAyaz) [inline api/images/images.router.js]:
```
res.json({image})
```

change in the whole file

File: client/src/app/admin/image-gallery/image-gallery.component.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  constructor(
+    private route: ActivatedRoute,
+    private dialog: MatDialog,
+    private eventService: EventsService,
+    private snackBar: MatSnackBar,
+    private imageService: ImageService
+    ) { }
+
+  ngOnInit() {
+    this.route.params.subscribe(params => {
```
Comment (AhsanAyaz) [inline client/src/app/admin/image-gallery/image-gallery.component.ts]: Unsubscribe the subscription on `ngOnDestroy`

File: client/src/app/admin/image-gallery/image-gallery.component.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  getEvent() {
+    this.eventService.getEventById(this.eventId)
+    .subscribe(
+    (data) => {
+      this.event = (data) ? data :  {};
+    },
+    (err) => {
+      this.showToast('Internal server Error');
```
Comment (AhsanAyaz) [inline client/src/app/admin/image-gallery/image-gallery.component.ts]: Do not show hard-coded errors, if there's an error, show the error content (whatever comes from the server

File: client/src/app/admin/image-gallery/image-gallery.component.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  getImages() {
+    this.imageService.get(this.eventId)
+    .subscribe(
+    (data: Image[]) => {
+      this.images = (data) ? data :  [];
+    },
+    (err) => {
+      this.showToast('Internal server Error');
```
Comment (AhsanAyaz) [inline client/src/app/admin/image-gallery/image-gallery.component.ts]: Do not show hard-coded errors, if there's an error, show the error content (whatever comes from the server

File: client/src/app/admin/image-gallery/image-gallery.component.ts [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  openDialog() {
+    const dialogRef = this.dialog.open(UploadImagesDialogComponent, {
+      data: {
+        entityId: this.eventId
+      },
+      width: '500px'
+    });
+    dialogRef.afterClosed().subscribe(result => {
```
Comment (AhsanAyaz) [inline client/src/app/admin/image-gallery/image-gallery.component.ts]: Either unsubscribe this on `ngOnDestroy`, or use 
```typescript
pipe(
  first()
)
.subscribe()
```

File: client/src/app/admin/image-gallery/image-gallery.component.ts [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+      if (result && result.upload) {
+          console.log(result);
```
Comment (AhsanAyaz) [inline client/src/app/admin/image-gallery/image-gallery.component.ts]: Remove console logs in the PRs

File: client/src/app/admin/image-gallery/image-gallery.component.ts [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+         this.images = result.upload.concat(this.images);
+      }
+    });
+  }
+
+  showToast(message) {
+    this.snackBar.open(message, '' , {
+      duration: 5000,
```
Comment (AhsanAyaz) [inline client/src/app/admin/image-gallery/image-gallery.component.ts]: Keep the duration of the messages to `1500`

File: client/src/app/core/models/event-item.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+export interface EventItem {
+  _id?: string;
+  name?: string;
+  description?: string;
+  organizationId?: string;
+  eventDate?: string;
+  createdAt?: string;
+// tslint:disable-next-line:eofline
```
Comment (AhsanAyaz) [inline client/src/app/core/models/event-item.ts]: Remove this tslint error comment and add a new line at the end of this file.

File: client/src/app/core/models/image.ts [hunk 11]
```diff
@@ -0,0 +1,6 @@
+export interface Image {
+  _id?: string;
+  entityId?: string;
+  path?: string;
+// tslint:disable-next-line:eofline
```
Comment (AhsanAyaz) [inline client/src/app/core/models/image.ts]: Remove this tslint error comment and add a new line at the end of this file.

File: client/src/app/core/services/events/events.service.ts [hunk 12]
```diff
@@ -58,4 +63,27 @@ export class EventsService {
...
+          filterEvent = event;
+          break;
+        }
+      }
+      return of(filterEvent);
+    } else if (this.events.length === 0) {
+      return this.http.get(`${this.apiBaseUrl}/event?_id=${eventId}`)
+        .pipe(
+          map((res) => {
+            return res['event'];
```
Comment (AhsanAyaz) [inline client/src/app/core/services/events/events.service.ts]: Remove usage of ['event'] and type the `http` call properly.

File: client/src/app/core/services/image/image.service.ts [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+  constructor(
+    private http: HttpClient
+  ) { }
+
+  get(entityId): Observable<Array<Image>> {
+    return this.http.get(`${this.apiBaseUrl}/images/all/${entityId}`)
+    .pipe(
+      map((res) => {
+        this.images = res['images'];
+        return res['images'];
```
Comment (AhsanAyaz) [inline client/src/app/core/services/image/image.service.ts]: Remove usage of `res['images']` and type the `http` call properly

File: client/src/app/shared/components/upload-images-dialog/upload-images-dialog.component.ts [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+import { ImageService } from '../../../core/services/image/image.service';
+
+@Component({
+  selector: 'em-upload-images-dialog',
+  templateUrl: './upload-images-dialog.component.html',
+  styleUrls: ['./upload-images-dialog.component.scss']
+})
+export class UploadImagesDialogComponent implements OnInit {
+  id: string;
+  imageList = [];
```
Comment (AhsanAyaz) [inline client/src/app/shared/components/upload-images-dialog/upload-images-dialog.component.ts]: Define a type (interface) for images and type the `imageList` property as:

```typescript
imageList: Array<ImageItem> = []
```

File: client/src/app/shared/components/upload-images-dialog/upload-images-dialog.component.ts [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+    public dialogRef: MatDialogRef<UploadImagesDialogComponent>,
+    private imageService: ImageService
+  ) { }
+
+  ngOnInit() {
+    this.id = this.data.entityId;
+  }
+
+  uploadDone(response) {
+    if (response.event.type === 4) {
```
Comment (AhsanAyaz) [inline client/src/app/shared/components/upload-images-dialog/upload-images-dialog.component.ts]: Combine the two `if` statements

File: api/event/event.model.js (hunk 1 above)
Comment (mehdirajani5) [reply api/event/event.model.js]: fixed.

File: api/images/images.router.js (hunk 2 above)
Comment (mehdirajani5) [reply api/images/images.router.js]: fixed

File: api/images/images.router.js (hunk 3 above)
Comment (mehdirajani5) [reply api/images/images.router.js]: fixed

File: client/src/app/admin/image-gallery/image-gallery.component.ts (hunk 5 above)
Comment (mehdirajani5) [reply client/src/app/admin/image-gallery/image-gallery.component.ts]: fixed

File: client/src/app/admin/image-gallery/image-gallery.component.ts (hunk 6 above)
Comment (mehdirajani5) [reply client/src/app/admin/image-gallery/image-gallery.component.ts]: fixed

File: client/src/app/admin/image-gallery/image-gallery.component.ts (hunk 4 above)
Comment (mehdirajani5) [reply client/src/app/admin/image-gallery/image-gallery.component.ts]: fixed

**Pull request merged**